File: lessons/5.2/3-find.js
const MongoClient = require('mongodb').MongoClient;

const uri = 'mongodb://localhost:27017';
const db = 'cooker';

(async function() {
  const client = new MongoClient(uri, { useUnifiedTopology: true});

  try {
    await client.connect();

    const database = client.db(db);

    const collection = database.collection('recipes');

    const recipes = collection.find();

    // iterate over the cursor
    while(await recipes.hasNext()) {
      const recipe = await recipes.next();
      console.dir(recipe.title);
    }

  } catch (err) {
    console.log(err.stack);
  }

  await client.close();
})();

File: lessons/5.4/3-find.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {

	client, err := mongo.NewClient(options.Client().ApplyURI("mongodb://localhost:27017"))

	if err != nil {
		log.Fatal(err)
	}

	context, _ := context.WithTimeout(context.Background(), 10*time.Second)
	err = client.Connect(context)

	if err != nil {
		log.Fatal(err)
	}

	defer client.Disconnect(context)

	type Recipe struct {
		Title string `bson:"title,omitempty"`
	}

	collection := client.Database("cooker").Collection("recipes")

	cur, currErr := collection.Find(context, bson.D{})

	if currErr != nil {
		panic(currErr)
	}
	defer cur.Close(context)

	var recipes []Recipe

	if err = cur.All(context, &recipes); err != nil {
		panic(err)
	}

	fmt.Println(recipes)

}
